pkg/operator/controller/ingress/deployment_test.go

```diff
@@ -327,6 +344,8 @@ func TestDesiredRouterDeployment(t *testing.T) {
 	checkDeploymentHasEnvVar(t, deployment, RouterEnableCompression, false, "")
 	checkDeploymentHasEnvVar(t, deployment, RouterCompressionMIMETypes, false, "")
 
+	checkDeploymentHasEnvSorted(t, deployment)
```

Comment:
Checking once at the end is sufficient.

Reply:
So that's what I did in the beginning; however, the bug actually was that env was not sorted specifically for EndpointPublishingStrategy = hostNetwork, though it was sorted for LoadBalancingService. So to be completely thorough, I placed it after every `desiredRouterDeployment` which generates a new ENV list. This test function is a bit overloaded as you might be able to tell.

Reply:
If you've got the time, it might be better to group all the env var checks together in a function that iterates over the list of env variable structs, applies them, then checks they are sorted at the end.  It would make this test easier to read and maintain, especially for new env variables.

Reply:
Yes I agree. I think this test function is a bit cluttered. Let me take a stab at cleaning some stuff up like you mentioned.

Reply:
I'm going to do some work on this for my current PR, so unless you've started, this lgtm.